sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ManifestFileCommitProtocol.scala

```diff
@@ -137,14 +139,7 @@ class ManifestFileCommitProtocol(jobId: String, path: String)
     if (addedFiles.nonEmpty) {
       val fs = new Path(addedFiles.head).getFileSystem(taskContext.getConfiguration)
       val statuses: Seq[SinkFileStatus] =
-        addedFiles.flatMap { f =>
```

Comment:
So, this if-else looked an overhead and it was added to avoid files not being written?

FWIW, there's still an old case when files are not written:

https://github.com/apache/spark/blob/73183b3c8c2022846587f08e8dea5c387ed3b8d5/sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala#L285-L297

```scala
spark.conf.set("spark.sql.orc.impl", "hive")
spark.range(10).filter(_ => false).write.orc("test.orc")
```

But I suspect it's a-okay since this behaviour will be superseded by "native" implementation completely in the future.

Reply:
Thanks for the pointer. That would also break streaming sink if there's empty partition then. Ideally it should be fixed for streaming query, but I feel its scope is beyond the PR.